lib/modules/manager/github-actions/community.ts

```diff
@@ -0,0 +1,118 @@
+import { z } from 'zod';
+
+import { escapeRegExp, regEx } from '../../../util/regex';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { NpmDatasource } from '../../datasource/npm';
+import { PypiDatasource } from '../../datasource/pypi';
+import * as condaVersioning from '../../versioning/conda';
+import * as npmVersioning from '../../versioning/npm';
+import * as pep440versioning from '../../versioning/pep440';
+
+import type { PackageDependency } from '../types';
+
+function matchFullUrl(domain: string, action: string): z.ZodString {
+  return z
+    .string()
+    .regex(
+      regEx(
+        escapeRegExp('https://' + domain + '/') +
+          escapeRegExp(action) +
+          '(@.+)?$',
+      ),
+    );
+}
+
+function matchAction(action: string): z.Schema {
+  return z.union([
+    z.string().regex(regEx(escapeRegExp(action) + '(@.+)?$')),
+    matchFullUrl('github.com', action),
+    matchFullUrl('code.forgejo.org', action),
+    matchFullUrl('data.forgejo.org', action),
+  ]);
+}
+
+/**
+ * schema here should match the whole step,
+ * there may be some actions use env as arguments version.
+ *
+ * each type should return `PackageDependency | undefined`
+ */
+export const communityActions = z.union([
+  z
+    .object({
+      // https://github.com/astral-sh/setup-uv
+      uses: matchAction('astral-sh/setup-uv'),
+      with: z.object({ version: z.string().refine((s) => s !== 'latest') }),
```

Comment:
what happens if the version isn't defined or `latest`? shouldn't cause an error debug message because of failed zod schema

Reply:
Fo you mean a skipped package with reason `unspecified-version`?

I'm not sure how to handle `latest` here becasue it's not a valid version spec but a special case of this action.

Reply:
Normally we'd pass it through as a "dep" but with either empty `currentValue` or set to `"latest"`. Then the versioning would take care of it and it would have `skipReason: "invalid-value"`.

Reply:
Yeah that's why I skip it in the first place, it's invalid in versioning but it's valid here.

Reply:
It can be good for users to see it in logs even if it's disabled (instead of silently dropping it)